src/core/painting/qgsgeometrypaintdevice.h

```diff
@@ -47,6 +47,22 @@ class QgsGeometryPaintEngine: public QPaintEngine
      */
     QgsGeometryPaintEngine( bool usePathStroker = false );
 
+    /**
+     * Sets the number of \a segments to use when drawing stroked paths with a rounded pen.
+     *
+     * The default is 8 segments, a smaller number will result in simpler paths.
```

Comment:
The default is 8 but we set 4 when rendering. Could we have the same value, either 4 ou 8 as default behavior?

Reply:
I'd rather leave this at the better quality default we use for buffering curves elsewhere (eg in QgsGeometry), as I've designed QgsGeometryPaintEngine around multiple use cases and not just mask rendering. For mask rendering we can get away with aggressive simplification, but this should not be the default.